Feature/dp 14175  stacked row section updates

merauluka opened the pull request:
## Description
This PR updates the application of the Stacked Row Section template and provides basic structural changes to consolidate css and add flexibility in when to apply certain classes to columned content.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-14175)
- [Related Drupal issue](https://jira.mass.gov/browse/DP-13168)

## Steps to Test
This updates the stacked row section wherever it is used, by primarily adjusting when and where padding is added. Verification can be performed by visiting the following pages in patternlab and noting they look as expected:
1. /?p=templates-services
2. /?p=pages-guide
3. /?p=pages-howto
4. /?p=pages-listing-links
5. /?p=pages-organization
6. /?p=pages-organization-elected-official

Commit:
DP-14175: Update to stacked row templates

* Updated styles on stacked row section to specifically target
that template.
* Updated templates to pull directly from stacked row section
organism to ensure future updates are reflected in templates.
* Add changelog file.

Commit:
DP-14175: Consolidated styles to remove conflicting classes

* Removed stacked-row in favor of stacked-row-section since it was called
in more places.
* Updated organism scss files to add in proper spacing for components.
* Updated templates to add in template-based overrides to organismas in
order to prevent unexpected spacing on pages that use the organism but
don't require the styling overrides.
* Updated markup as needed on source templates to ensure styles are
applied for stacked rows pages as expected.

ygannett commented on assets/scss/03-organisms/_action-finder.scss:
```diff
@@ -184,6 +184,10 @@ $action-finder-bp: 900px;
     text-align: left;
   }
+
+  .ma__stacked-row__section .main-content .page-content > & {
```
See my comment for _divider.scss.

ygannett commented on assets/scss/03-organisms/_event-listing.scss:
```diff
@@ -11,6 +11,10 @@ $event-listing-bp-show-grid: $bp-medium-min;
     }
   }
 
+  .ma__stacked-row__section .main-content .page-content > & {
```
See my comment for _divider.scss.

ygannett commented on assets/scss/03-organisms/_key-actions.scss:
```diff
@@ -1,6 +1,10 @@
 .ma__key-actions {
   clear: both;
 
+  .main-content.main-content--full .page-content > & {
```
See my comment for _stacked-row.scss.

ygannett commented on assets/scss/03-organisms/_mapped-locations.scss:
```diff
@@ -56,4 +56,9 @@
     margin-top: -0.5rem;
   }
+
+  .ma__stacked-row__section .main-content .page-content > & {
```
See my comment for _stacked-row.scss.

ygannett commented on assets/scss/04-templates/_organization.scss:
```diff
@@ -0,0 +1,2 @@
+.ma__organization {
```
Are you intentionally leaving this empty?

ygannett commented on assets/scss/01-atoms/_divider.scss:
```diff
@@ -2,6 +2,10 @@
   @include ma-container;
   border: none;
 
+  .ma__stacked-row__section .main-content .page-content > & {
```
I wonder this block can go to **assets/scss/04-templates/_stacked-row.scss** as a child element of `.page-content` since it looks a specific style when it's under a stack row. 
Then, it's not necessary to be added repeatedly to multiple places below scss files unless it needs a component specific control. It's easier to maintain that way, I think.

ygannett commented on assets/scss/03-organisms/_stacked-row-section.scss:
```diff
@@ -1,70 +1,57 @@
-  }
-}
-
-//theme
```
Theme specific styles are separated from the common styles, so everybody knows where to touch for theme related changes.

merauluka replied:
Based on my conversations with @clair0917 combining the theme sections was desirable. If that is not the case I can put these back. It just seemed odd because these were often declaring the same thing that was already set above.

ygannett commented on assets/scss/04-templates/_details.scss:
```diff
@@ -29,19 +29,8 @@
-  }
-}
-
-//theme
```
Theme specific styles are separated from the common styles, so everybody knows where to touch for theme related changes.

ygannett replied:
The same suggestion as the `assets/scss/03-organisms/_stacked-row-section.scss`'s.

- Remove LL. 34-35 `}`.
- Keep `//theme` at L. 37.
- Remove LL. 38-41
   ```
   .ma__details {
   
     .page-content > .ma__contact-list {
       
   ```
- Add `//theme` to L. 47

ygannett commented on assets/scss/04-templates/_information-details.scss:
```diff
@@ -1,13 +1,3 @@
-    }
-  }
-}
-//theme
```
Theme specific styles are separated from the common styles, so everybody knows where to touch for theme related changes.

ygannett replied:
The same suggestion as the `assets/scss/03-organisms/_stacked-row-section.scss`'s.
This one appears to be all styles are for theming.

- Remove LL. 3-9.
   ```
   &__content {

       & > * {
         @include ma-component-spacing;
       }
     }
   }
   ```
- Keep L. 10`//theme`.
- Remove LL. 11-12.
   ```
   .ma__information-details {

   ```

ygannett reviewed (commented):
Please see my comments.

Commit:
DP-14175: Consolidated theme changes

Updated css changes so they are more logically grouped when
they are applying overrides.

Commit:
DP-14175: Update image credit override padding

Commit:
DP-14175: Update theme styles

Restored theme styles that were removed.

ygannett reviewed (approved):
LGTM.

Commit:
DP-14175: Add option to allow setting a class on main-content div

Commit:
Merge branch 'develop' into feature/DP-14175--stacked-row-section-updates

Commit:
DP-14175: Correct border color application

Commit:
DP-14175: Update to resolve errors after latest merge

Commit:
DP-14175: Update stacked row class after merging from develop

mrossi113 commented:
The Drupal PR passed IR so I'm merging this PR into develop now.
